mlflow/projects/__init__.py

```diff
@@ -319,10 +328,20 @@ def run(
     elif env_manager is not None:
         _EnvManager.validate(env_manager)
 
+    _has_valid_tracking_uri = (
+        tracking_uri is not None
+    )  # Prevent having to do the check manually below,
+    # and can be replaced with a function that does the check
+
     if backend == "databricks":
-        mlflow.projects.databricks.before_run_validations(mlflow.get_tracking_uri(), backend_config)
-    elif backend == "local" and run_id is not None:
-        backend_config_dict[MLFLOW_LOCAL_BACKEND_RUN_ID_CONFIG] = run_id
+        _databricks_uri = tracking_uri if _has_valid_tracking_uri else mlflow.get_tracking_uri()
+        mlflow.projects.databricks.before_run_validations(_databricks_uri, backend_config)
+    else:
+        if backend == "local" and run_id is not None:
+            backend_config_dict[MLFLOW_LOCAL_BACKEND_RUN_ID_CONFIG] = run_id
+        if _has_valid_tracking_uri:
+            mlflow.set_tracking_uri(tracking_uri)
+            # prevent setting tracking_uri to the env MLFLOW_TRACKING_URI
```

Comment:
Now that we're passing `tracking_uri` to `_run()`, I think we can safely remove `mlflow.set_tracking_uri()` and the associated `if _has_valid_tracking_uri` check

Reply:
Can we also update the Project execution logic for Kubernetes to use the  specified tracking URI? https://github.com/mlflow/mlflow/blob/c20c664e037b895ff4bd49dda4a26604e55a9b93/mlflow/projects/__init__.py#L139-L179. Setting the `MLFLOW_TRACKING_URI` environment variable seems like a good approach, similar to your approach for Databricks support.

Reply:
I will have a look 👍🏻